packages/jest-cli/src/TestPathPatternModeController.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+
+'use strict';
+
+import type {HasteContext} from 'types/HasteMap';
+import type {Config, Path} from 'types/Config';
+
+const ansiEscapes = require('ansi-escapes');
+const chalk = require('chalk');
+const {getTerminalWidth} = require('./lib/terminalUtils');
+const highlight = require('./lib/highlight');
+const stringLength = require('string-length');
+const {trimAndFormatPath} = require('./reporters/utils');
+const SearchSource = require('./SearchSource');
+const PromptController = require('./lib/PromptController');
+
+const pluralizeFile = (total: number) => total === 1 ? 'file' : 'files';
+
+const usage = (delimiter: string = '\n') => {
+  const messages = [
+    `\n ${chalk.bold('Pattern Mode Usage')}`,
+    ` ${chalk.dim('\u203A Press')} ESC ${chalk.dim('to exit pattern mode.')}\n`,
+  ];
+
+  return messages.filter(message => !!message).join(delimiter) + '\n';
+};
+
+const usageRows = usage().split('\n').length;
+
+const testPathPatternModeController = (
+  config: Config,
+  pipe: stream$Writable | tty$WriteStream,
+  promptController: PromptController,
+) => {
+  class TestPathPatternModeController {
+    searchSource: SearchSource;
+
+    run(
+      onSuccess: Function,
+      onCancel: Function,
+    ) {
+      pipe.write(ansiEscapes.cursorHide);
+      pipe.write(ansiEscapes.clearScreen);
+      pipe.write(usage());
+      pipe.write(ansiEscapes.cursorShow);
+
+      promptController.prompt(
+        this.onChange.bind(this),
```

Comment:
We can bind this in the constructor or use class property syntax so that we don't have to re-bind it every time that it runs.

bind in constructor

```javascript
constructor() {
  this.onChange = this.onChange.bind(this);
}
```

class property
```javascript
onChange = () => {
// ...content
}
```